### PR TITLE
Merge grake

### LINKED ISSUE #36: [CRAN preparation] merge grake's functionalities into MultiLevelIPF
ref: https://github.com/krlmlr/MultiLevelIPF/pull/35#issuecomment-808843398

@krlmlr should users be able to use grake's functions in MultiLevelIPF or is it only for `ml_fit_dss()` to work without an external dependency?

### PR DESCRIPTION
**Pull request by asiripanich**
This PR closes #36.

- copied over `dss()` and `gginv()`.
- copied over all tests.
- removed krlmlr::grake from DESCRIPTION's remote and import fields.
- exported `dss()` and `gginv()`

### CODE REVIEW

**Comment by asiripanich** [comment]
Failing checks
- rcc: `Note: Package unavailable to check Rd xrefs: 'sampling'`
- pkgdown: `Error: Error: <callr_status_error: callr subprocess failed: Topics missing from index: dss`

File: DESCRIPTION [hunk 1]
```diff
@@ -27,8 +26,8 @@ Imports:
 Suggests: 
     covr,
     testthat,
+    MASS,
```
Comment (krlmlr) [inline DESCRIPTION:29]:
```suggestion
    MASS,
    sampling,
```

**Review by krlmlr — COMMENTED** [review]
Looks good, thanks! Should we add the authors of `gginv()` and `dss()` to `DESCRIPTION`?

**Comment by krlmlr** [comment]
Please also add yourself if you like.

**Comment by asiripanich** [comment]
Sure I will add the other authors and myself. Thanks @krlmlr !

**Comment by asiripanich** [comment]
> Looks good, thanks! Should we add the authors of `gginv()` and `dss()` to `DESCRIPTION`?

We should acknowledge them appropriately, but I'm not sure if we should include them in DESCRIPTION without their knowledge?

**Comment by asiripanich** [comment]
By the way, are you interested in submitting the package to the Journal of Open Source Software (JOSS) ? I think it would be great if the package can be cited as a JOSS paper :) 

I'm happy to help with the submission etc.

**Comment by krlmlr** [comment]
Thanks! I'll let the CRAN wizards guide me to the correct attribution.

Let's discuss JOSS elswhere.

**Comment by asiripanich** [comment]
Sounds great. What's your email?

**Comment by krlmlr** [comment]
I sent you one.

**Pull request merged**